src/components/Checkbox/Checkbox.tsx

```diff
@@ -56,8 +60,15 @@ const Checkbox: FC<CheckboxProps> = props => {
         );
     }
 
+    const checkboxRef = React.createRef<HTMLInputElement>();
+
+    useEffect(() => {
```

Comment:
What is the useEffect in combination with the ref accomplishing here? I would assume that you would be able to pass the prop directly to the checkbox

Reply:
Currently, browsers don't support setting this through the HTML, and must be set though javascript
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#indeterminate

Reply:
Thanks for the explanation. Can you add this comment to the code as well and explain why it is the way it is? That way future devs aren't confused

Reply:
Done!